ironfish-cli/src/commands/chain/rewind.ts

```diff
@@ -136,15 +136,17 @@ async function rewindWalletHead(
       })
       speed.start()
 
-      wallet.chainProcessor.onRemove.on((_) => {
-        speed.add(1)
-        bar.update(++rewound, {
-          speed: speed.rate1s.toFixed(2),
-          estimate: TimeUtils.renderEstimate(rewound, toRewind, speed.rate1m),
+      const scan = await wallet.scanner.scan()
+
+      if (scan) {
+        scan.onTransaction.on((_) => {
+          speed.add(1)
+          bar.update(++rewound, {
+            speed: speed.rate1s.toFixed(2),
+            estimate: TimeUtils.renderEstimate(rewound, toRewind, speed.rate1m),
+          })
         })
-      })
-
-      await wallet.updateHead()
+      }
```

Comment:
May need to rethink this

Reply:
Maybe this should use the sequence, similar to what `rescan` does?